From 15b3a8fc8635e04fa7ea1275b910772d8f2820a2 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Sun, 10 May 2020 02:44:47 +0200
Subject: [PATCH] always default to qtquick software backend on big endian

the default backend is buggy and breaks rendering of various
components of KDE Plasma and LXQt, so just default to software
so the user doesn't have to specify QT_QUICK_BACKEND
---
 qtdeclarative/src/quick/scenegraph/qsgcontextplugin.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git qtdeclarative/src/quick/scenegraph/qsgcontextplugin.cpp qtdeclarative/src/quick/scenegraph/qsgcontextplugin.cpp
index 66add51..a0fdaa2 100644
--- a/qtdeclarative/src/quick/scenegraph/qsgcontextplugin.cpp
+++ b/qtdeclarative/src/quick/scenegraph/qsgcontextplugin.cpp
@@ -139,7 +139,10 @@ QSGAdaptationBackendData *contextFactory()
         // If this platform does not support OpenGL, and no backend has been set
         // default to the software renderer
         if (requestedBackend.isEmpty()
-            && !QGuiApplicationPrivate::platformIntegration()->hasCapability(QPlatformIntegration::OpenGL)) {
+#if Q_BYTE_ORDER == Q_LITTLE_ENDIAN
+            && !QGuiApplicationPrivate::platformIntegration()->hasCapability(QPlatformIntegration::OpenGL)
+#endif
+        ) {
             requestedBackend = QString::fromLocal8Bit("software");
         }
 
-- 
2.25.1

